src/flightctl/context.rs

```diff
@@ -1,141 +1,208 @@
+        password: None,
+        token: None,
+        token_file: None,
+        username: None,
+    };
+
     log::debug!("Configuring credentials: {:?}", auth);
     match auth.config {
         AuthConfig::AwsSso { .. } => match &cluster.config {
             ClusterConfig::Eks { name, region } => {
                 log::info!(
-                    "Setting Kubernetes credentials for EKS cluster: {} as {} in {}",
+                    "Using AWS profile {} for EKS cluster {} in region {}",
+                    auth.name,
                     name,
-                    context.name,
                     region
                 );
-                kubeconfig::create_auth(
-                    &context.name,
-                    &[
-                        "--exec-api-version",
-                        "client.authentication.k8s.io/v1alpha1",
-                        "--exec-arg",
-                        "--region",
-                        "--exec-arg",
-                        region,
-                        "--exec-arg",
-                        "eks",
-                        "--exec-arg",
-                        "get-token",
-                        "--exec-arg",
-                        "--cluster-name",
-                        "--exec-arg",
-                        name,
-                        "--exec-command",
-                        "aws",
-                        "--exec-env",
-                        &format!("AWS_PROFILE={}", auth.name),
-                    ],
-                )?;
-                log::debug!("Credentials configured");
-                Ok(())
+
+                let mut env = HashMap::new();
+                env.insert(String::from("name"), String::from("AWS_PROFILE"));
+                env.insert(String::from("value"), auth.name.clone());
+
+                result.exec = Some(ExecConfig {
+                    api_version: Some(String::from("client.authentication.k8s.io/v1alpha1")),
+                    args: Some(vec![
+                        String::from("--region"),
+                        region.to_string(),
+                        String::from("eks"),
+                        String::from("get-token"),
+                        String::from("--cluster-name"),
+                        name.to_string(),
+                    ]),
+                    command: String::from("aws"),
+                    env: Some(vec![env]),
+                })
             }
         },
     }
+
+    NamedAuthInfo {
+        name: context.name.clone(),
+        auth_info: result,
+    }
 }
 
-fn ensure_cluster(context: &Context, config: &Config) -> anyhow::Result<()> {
+fn ensure_cluster(config: &Kubeconfig, expected: &NamedCluster) -> anyhow::Result<()> {
     log::debug!("Checking for Kubernetes cluster");
-    let exists = kubeconfig::cluster_exists(&context.cluster)?;
+
+    let exists = config
+        .clusters
+        .iter()
+        .find(|actual| {
+            &actual.name == &expected.name
```

Comment:
For this, and some of the other comparison operations, are these (`actual` and `expected`) the same types? If so, does it make sense to implement a flavor of `Eq` separately to slim these down?

Reply:
These types come from [kube::config](https://docs.rs/kube/0.16.1/kube/config/index.html), and they don't implement `Eq`.